sirepo/template/srw_shadow.py

```diff
@@ -781,6 +782,13 @@ def __sim_to_srw(self, shadow, srw):
         srw.simulation.distanceFromSource = shadow.plotXYReport.distanceFromSource
         srw.initialIntensityReport.colorMap = shadow.initialIntensityReport.colorMap
 
+    def __size_srw_mirror(self, srw_mirror, shadow_mirror):
```

Comment:
Still not quite right - if fhit_c is 0 the shadow mirror is infinite, so the 10m value should be used - none of the other values can be trusted. If fhit_c is 1 (finite) then use fshape to determine which fields to translate (rectangular use halfWidthX1, halfWidthX2, halfLengthY1, halfLengthY2), (ellipse or ellipse with hole use externalOutlineMajorAxis, externalOutlineMinorAxis)

Otherwise the SRW sim will fail if an infinite mirror is selected and all the shadow mirror sizes are 0.